cmd/dump/dump.go

```diff
@@ -40,17 +43,15 @@ func main() {
 	for {
 		record := iterator.Record()
 
-		keyReader, keySize := record.Key()
-		key := make([]byte, keySize)
-		if _, err = keyReader.Read(key); err != nil {
-			log.Fatal(err)
-		}
+		keyReader, _ := record.Key()
+		keyBuf := new(bytes.Buffer)
```

Comment:
I don't think that this approach could help to optimize memory usage. Every loop step you allocate a bytes.Buffer object and io.Copy also allocates []byte. I think such code as 

```go
  var (
         key   = []byte{} // better to allocate with some capacity
         value = []byte{}
  )
  ....
  loop step
  ....

  if len(key) < int(keySize) {
       key = make([]byte, keySize*2)
  } // otherwise we save memory for key reading

  // the same for value reading
  ...
  
  err = csvWriter.Write([]string{string(key[:keySize]), string(value[:valSize])})
```

could help in the issue.